src/server/dflycmd.cc

```diff
@@ -413,8 +415,10 @@ void DflyCmd::TakeOver(CmdArgList args, ConnectionContext* cntx) {
   absl::Time start = absl::Now();
   AggregateStatus status;
 
-  // We need to await for all dispatches to finish: Otherwise a transaction might be scheduled
-  // after this function exits but before the actual shutdown.
+  // We need to wait for all dispatches to finish to not interrupt any ongoing transactions.
+  // TODO: Currently we just wait for a moment during which no dispatches are going on
+  // (guaranteeing that all previous finished), however such a moment might take a while to
+  // catch under heavy load - our goal is different!
```

Comment:
"our goal is different"  - do not leave us in suspense, what's our goal? :)

Reply:
I reverted the comment because I'm fixing this issue in #2156